pkg/process/metadata/workloadmeta/workloadmeta.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// ProcessEntity is a placeholder workloadmeta.ProcessEntity.
+// It does not contain all the fields that the final entity will contain.
+type ProcessEntity struct {
+	pid      int32
+	cmdline  []string
+	language *languagemodels.Language
+}
+
+// WorkloadMetaExtractor handles enriching processes with
+type WorkloadMetaExtractor struct {
+	cache      map[string]*ProcessEntity
+	cacheMutex sync.RWMutex
+
+	grpcListener mockableGrpcListener
+}
+
+// NewWorkloadMetaExtractor constructs the WorkloadMetaExtractor.
+func NewWorkloadMetaExtractor() *WorkloadMetaExtractor {
+	return &WorkloadMetaExtractor{
```

Comment:
What do you think of emitting a Debug log here so we can track when a new `WorkloadMetaExtractor` is created?

Reply:
I'm not opposed to it, but why? At the moment it's only instantiated in one place. I am going to add a `Start` function that starts the grpc server, maybe that's a better place for it?

Reply:
Discussed offline, this is useful for QA to make sure that the config setting is being respected.